src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -302,7 +306,14 @@ public Completable start() {
         negotiate.flatMapCompletable(url -> {
             logger.debug("Starting HubConnection.");
             if (transport == null) {
-                transport = new WebSocketTransport(headers, httpClient);
+                switch (transportEnum){
+                    case LONG_POLLING:
+                        this.httpClient = new DefaultHttpClient(this.pollTimeout);
+                        transport = new LongPollingTransport(headers, httpClient);
```

Comment:
Why are headers passed to the transport ctor? This makes the accessTokenProvider useless if you don't use the TransportEnum. I get that you could configure the Authorization header yourself if you supply your own transport, but I think it might be better to have an API on the transport to change the headers dynamically.

Reply:
Currently  the Transport interface is private/pubternal across all our clients so we generally  expect users to use the enum to select their transport.